### PR TITLE
buildah: correctly create the userns if euid!=0

### PR DESCRIPTION
**Pull request by giuseppe**
we were skipping the creation of the creation of a namespace if the
euid != 0.

regression introduced by 3d7403130100c3558ea14a1c21482559109de043

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
LGTM, thanks @giuseppe. Are missing rootless buildah tests?

@TomSweeneyRedHat @rhatdan @nalind PTAL

File: cmd/buildah/unshare.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func bailOnError(err error, format string, a ...interface{}) {
 
 func maybeReexecUsingUserNamespace(cmdName string, evenForRoot bool) {
 	// If we've already been through this once, no need to try again.
-	if unshare.IsRootless() {
+	if os.Geteuid() == 0 && unshare.IsRootless() {
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/unshare.go]: should this be '!= 0'?  If so,isn't that covered in unshare.IsRootless already?

 		isRootless = os.Geteuid() != 0 || os.Getenv(UsernsEnvName) != ""

Comment (giuseppe) [reply cmd/buildah/unshare.go]: no, it should be == 0, we want to skip the test here only if we are already running as root user.

Inside unshare.IsRootless then we might return early if the env variable is set

Comment (TomSweeneyRedHat) [reply cmd/buildah/unshare.go]: OK, the title for the PR threw me off, just a a dyslexic moment on my end. I see what's happening now.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please.

**Comment by giuseppe** [comment]
please wait to merge, there are other fixes needed

**Comment by TomSweeneyRedHat** [comment]
@giuseppe, if tests pass is this good to merge?

**Comment by rhatdan** [comment]
LGTM

**Comment by giuseppe** [comment]
it is ready to merge now

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**